src/super_gradients/training/utils/callbacks.py

```diff
@@ -765,3 +768,26 @@ def __init__(self, lr_placeholder):
 
     def __call__(self, context: PhaseContext):
         self.lr_placeholder.append(context.optimizer.param_groups[0]["lr"])
+
+
+class CallbackNames:
+    DECI_LAB_UPLOAD = 'DeciLabUploadCallback'
+    LR_CALLBACK_BASE = 'LRCallbackBase'
+    LR_SCHEDULER = 'LRSchedulerCallback'
+    METRICS_UPDATE = 'MetricsUpdateCallback'
+    MODEL_CONVERSION_CHECK = 'ModelConversionCheckCallback'
+    EARLY_STOP = 'EarlyStop'
+    DETECTION_MULTISCALE_PREPREDICTION = 'DetectionMultiscalePrePredictionCallback'
+    YOLOX_TRAINING_STAGE_SWITCH = 'YoloXTrainingStageSwitchCallback'
+
+
+CALLBACKS = {
+    CallbackNames.DECI_LAB_UPLOAD: DeciLabUploadCallback,
+    CallbackNames.LR_CALLBACK_BASE: LRCallbackBase,
+    CallbackNames.LR_SCHEDULER: LRSchedulerCallback,
+    CallbackNames.METRICS_UPDATE: MetricsUpdateCallback,
+    CallbackNames.MODEL_CONVERSION_CHECK: ModelConversionCheckCallback,
+    CallbackNames.EARLY_STOP: EarlyStop,
+    CallbackNames.DETECTION_MULTISCALE_PREPREDICTION: DetectionMultiscalePrePredictionCallback,
+    CallbackNames.YOLOX_TRAINING_STAGE_SWITCH: YoloXTrainingStageSwitchCallback
+}
```

Comment:
I did the same on callbacks to have a unified approach but I got an error when importing EarlyStop in the CI.
I think this is due to circular import, and to avoid it I think we should do like with optimizers, having a folder including 2 modules, one with the functions and one with what I just added.

What do you think ? Should we just leave this for later and not modify the callback factory, or should we do the refacto I talked about ? Or any other option ?

Reply:
https://github.com/Deci-AI/super-gradients/blob/bfaaefbc25003f0abc97186aa561d53b8471d47f/src/super_gradients/training/utils/callbacks.py#L773-L793